### PR TITLE
update kubernetes client library package

### PR DESCRIPTION
**Pull request by jmuk**
This is similar to https://github.com/istio/pilot/pull/966, updating k8s.io/client-go and its dependencies. adapters/kubernetes also needs to update some module name due to this update.

---
This change is [Reviewable](https://reviewable.io/reviews/istio/mixer/975)

### CODE REVIEW

**Comment by douglas-reid** [comment]
How does this compare to #963  ?

**Comment by jmuk** [comment]
oh, didn't notice that. Closing.

File: adapter/kubernetes/cache.go [hunk 1]
```diff
@@ -20,10 +20,11 @@ import (
 	"reflect"
 	"time"
 
+	"k8s.io/api/core/v1"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
```
Comment (douglas-reid) [inline adapter/kubernetes/cache.go]: can we just use `metav1` instead (no underscore)?

**Comment by jmuk** [comment]
reopened: #963 is referring to a bit old version of client-go. This refers to the same revision as istio/pilot#966

File: adapter/kubernetes/cache.go (hunk 1 above)
Comment (jmuk) [inline adapter/kubernetes/cache.go]: done

**Pull request merged**